### PR TITLE
Update spectral rule to read severity from spectral result

### PR DESCRIPTION
**Pull request by niclim**
## 🍗 Description
_What does this PR do? Anything folks should know?_

Updates spectral rule to read from spectral severity

Based off https://github.com/opticdev/optic/pull/1829

## 📚 References
_Links to relevant docs (Notion, Twist, GH issues, etc.), if applicable._

## 👹 QA
_How can other humans verify that this PR is correct?_

### CODE REVIEW

**Review by acunniffe — APPROVED** [review]

File: projects/rulesets-base/src/extended-rules/spectral-rule.ts [hunk 1]
```diff
@@ -54,6 +64,15 @@ function toOpticResult(
   };
 }
 
+// This will treat info + hint as a info in Optic
+function severityToOpticSeverity(spectralSev: DiagnosticSeverity): Severity {
+  return spectralSev === DiagnosticSeverity.Error
```
Comment (acunniffe) [inline projects/rulesets-base/src/extended-rules/spectral-rule.ts:69]: looks good. If I'm reading this right it looks like we only support `error` `warn` and `info`? seems reasonable.

Comment (niclim) [reply projects/rulesets-base/src/extended-rules/spectral-rule.ts:69]: Yes thats correct

**Comment by niclim** [comment]
There's a bug here that I need to fix, spectral rules don't appear to be running correctly on added in certain cases - need to spend some time to figure out and fix this

File: projects/rulesets-base/src/extended-rules/spectral-rule.ts [hunk 2]
```diff
@@ -186,8 +205,21 @@ export class SpectralRule extends ExternalRuleBase {
...
           changesByJsonPath[fact.location.jsonPath];
+
+        if (
+          !maybeChange &&
+          jsonPointerHelpers.matches(fact.location.jsonPath, [
+            'paths',
+            '**',
+            `{${Object.values(OpenAPIV3.HttpMethods).join(',')}}`,
+          ])
+        ) {
```
Comment (niclim) [inline projects/rulesets-base/src/extended-rules/spectral-rule.ts:218]: @acunniffe just an FYI, this is kind of weird but it's the same thing as the groupDiffs w/ path.

There's a few solutions to fixing this, one of which is some sort of internal spec format, the other being a better abstraction around the fact trees / jsonPath -> Fact type

**Pull request merged**